python/cuml/benchmark/bench_helper_funcs.py

```diff
@@ -44,8 +44,17 @@ def _build_fil_classifier(m, data, arg={}):
         raise ImportError("No XGBoost package found")
 
     # use maximum 1e5 rows to train the model
-    train_size = min(data[0].shape[0], 100000)
-    dtrain = xgb.DMatrix(data[0][:train_size, :], label=data[1][:train_size])
+    train_size = min(data[0].shape[0], 1000)
+
+    if isinstance(data[0], np.ndarray):
+        dtrain = xgb.DMatrix(data[0][:train_size, :], label=data[1][:train_size])
+    else:
+        # if the input is gpuarray
```

Comment:
You can use `input_to_dev_array` here if you want to accept any gpu array format compatible with cuda_array_interface.

Otherwise it seems you're expecting a numba or rmm device_array (instead of cupy or pytorch or whatever, presence of `copy_to_host`), you might want to add a check here with `cuda.devicearray.is_cuda_ndarray(data)` (need to add `from numba import cuda`).

Reply:
Thanks Dante, this is really helpful!